docs/development/CodingStyle.md

```diff
@@ -0,0 +1,259 @@
+
+If it is followed by an `else` or `else if` that shall be on the same line, again with the opening brace on the same line.
+
+A single statement after an `if` or an `else` may omit the "unnecessary" braces only when ALL conditional branches have single statements AND you have strong reason to know it will always be that way.
+
+If in doubt, do not omit such "unnecessary" braces.
+(Adding a statement to a branch will break the logic if the braces are forgotten and otherwise make the PR longer).
+
+##Spaces
+Use a space after (most) keywords.  The notable exceptions are sizeof, typeof, alignof, and __attribute__, which look somewhat like functions (and are usually used with parentheses).
+So use a space after these keywords:
+```
+if, switch, case, for, do, while
+```
+but not with sizeof, typeof, alignof, or __attribute__.  E.g.,
+```
+s = sizeof(struct file);
+```
+When declaring pointer data or a function that returns a pointer type, the preferred use of '*' is adjacent to the data name or function name and not adjacent to the type name.  Examples:
+```
+char *linux_banner;
+memparse(char *ptr, char **retptr);
+char *match_strdup(substring_t *s);
+```
+Use one space around (on each side of) most binary and ternary operators, such as any of these:
+```
+=  +  -  <  >  *  /  %  |  &  ^  <=  >=  ==  !=  ?  :
+```
+but no space after unary operators:
+```
+&  *  +  -  ~  !  sizeof  typeof  alignof  __attribute__  defined
+```
+no space before the postfix increment & decrement unary operators:
+```
+++  --
+```
+no space after the prefix increment & decrement unary operators:
+```
+++  --
+```
+and no space around the '.' and "->" structure member operators.
+
+'*' and '&', when used for pointer and reference, shall have no space between it and the following variable name.
+
+#typedef
+enums with a count should have that count declared as the last item in the enumeration list,
+so that it is automatically maintained, e.g.:
+```
+typedef enum {
+    PID_CONTROLLER_MW23 = 0,
+    PID_CONTROLLER_MWREWRITE,
+    PID_CONTROLLER_LUX_FLOAT,
+    PID_COUNT
+} pidControllerType_e;
+```
+It shall not be calculated afterwards, e.g. using PID_CONTROLLER_LUX_FLOAT + 1;
+
+typedef struct definitions should include the struct name, so that the type can be forward referenced, that is in:
+```
+typedef struct motorMixer_s {
+    float throttle;
+...
+    float yaw;
+} motorMixer_t;
+```
+the motorMixer_s name is required.
+
+#Variables
+
+##Naming
+Generally, descriptive lowerCamelCase names are preferred for function names, variables, arguments, etc.
+For configuration variables that are user accessible via CLI or similar, all_lowercase with underscore is preferred.
+
+Variable names should be nouns.
+
+Simple temporary variables with a very small scope may be short where it aligns with common practice.
+Such as "i" as a temporary counter in a `for` loop, like `for (uint_8 i = 0; i < 4; i++)`.
```

Comment:
the`uint_8` in this example is suboptimal on ARM + gcc, using `int` will produce better code and is cleaner